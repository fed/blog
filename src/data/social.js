import twitterIcon from '../assets/twitter.svg';
import githubIcon from '../assets/github.svg';
import mediumIcon from '../assets/medium.svg';
import linkedinIcon from '../assets/linkedin.svg';
import codepenIcon from '../assets/codepen.svg';

export default [
    {
        id: 'twitter',
        name: 'Twitter',
        url: 'https://twitter.com/fknussel',
        icon: twitterIcon
    },
    {
        id: 'github',
        name: 'GitHub',
        url: 'https://github.com/fed',
        icon: githubIcon
    },
    {
        id: 'medium',
        name: 'Medium',
        url: 'https://medium.com/@fknussel',
        icon: mediumIcon
    },
    {
        id: 'codepen',
        name: 'CodePen',
        url: 'https://codepen.io/fede',
        icon: codepenIcon
    },
    {
        id: 'linkedin',
        name: 'LinkedIn',
        url: 'https://linkedin.com/in/fknussel',
        icon: linkedinIcon
    }
];
