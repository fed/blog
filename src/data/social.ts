import codepenIcon from '../assets/codepen.svg';
import githubIcon from '../assets/github.svg';
import linkedinIcon from '../assets/linkedin.svg';
import mediumIcon from '../assets/medium.svg';
import twitterIcon from '../assets/twitter.svg';
import { Social } from '../types';

export const socialLinks = [
    {
        id: Social.TWITTER,
        name: 'Twitter',
        url: 'https://twitter.com/fknussel',
        icon: twitterIcon,
    },
    {
        id: Social.GITHUB,
        name: 'GitHub',
        url: 'https://github.com/fed',
        icon: githubIcon,
    },
    {
        id: Social.MEDIUM,
        name: 'Medium',
        url: 'https://medium.com/@fknussel',
        icon: mediumIcon,
    },
    {
        id: Social.CODEPEN,
        name: 'CodePen',
        url: 'https://codepen.io/fede',
        icon: codepenIcon,
    },
    {
        id: Social.LINKEDIN,
        name: 'LinkedIn',
        url: 'https://linkedin.com/in/fknussel',
        icon: linkedinIcon,
    },
];
