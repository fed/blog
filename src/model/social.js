import twitterIcon from '../assets/twitter.svg';
import githubIcon from '../assets/github.svg';
import mediumIcon from '../assets/medium.svg';
import linkedinIcon from '../assets/linkedin.svg';

export default [
  {
    id: 'twitter',
    name: 'Twitter',
    url: '//twitter.com/fknussel',
    icon: twitterIcon
  },
  {
    id: 'github',
    name: 'GitHub',
    url: '//github.com/fknussel',
    icon: githubIcon
  },
  {
    id: 'medium',
    name: 'Medium',
    url: '//medium.com/@fknussel',
    icon: mediumIcon
  },
  {
    id: 'linkedin',
    name: 'LinkedIn',
    url: '//linkedin.com/in/fknussel',
    icon: linkedinIcon
  }
];
