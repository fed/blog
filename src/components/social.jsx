import React from 'react';
import styled from 'styled-components';

import { socialLinks } from '../data';
import { Button } from './button';

// Wanted to give the container a 30px padding all around.
// Since the buttons have a vertical drop shadow of 6px, this means the bottom padding is 36px.
// But all buttons have a bottom margin of 18px in case they get stacked,
// meaning the container's bottom padding is: 30px + 6px - 18px = 18px

const Container = styled.nav`
    background-color: #f5f5f5;
    margin: 48px 0;
    padding: 30px 30px 18px;
    text-align: center;
    @media (min-width: 1024px) {
        background-color: transparent;
        padding: 0;
    }
`;

const Link = styled(Button)`
    margin-bottom: 18px;
    text-align: center;
    :not(:last-child) {
        margin-right: 10px;
    }
`;

export function Social() {
    return (
        <Container aria-label="Social media links">
            {socialLinks.map(item => (
                <Link key={item.id} type={item.id} url={item.url}>
                    {item.name}
                </Link>
            ))}
        </Container>
    );
}
