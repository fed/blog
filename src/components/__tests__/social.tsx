import React from 'react';
import renderer from 'react-test-renderer';

import { Social } from '../social';

describe('Social', () => {
    it('renders correctly', () => {
        const tree = renderer.create(<Social />).toJSON();

        expect(tree).toMatchSnapshot();
    });
});
