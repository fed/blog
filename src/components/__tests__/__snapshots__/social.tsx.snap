// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Social renders correctly 1`] = `
<nav
  aria-label="Social media links"
  className="sc-gKckTs cELkVU"
>
  <a
    className="sc-bdvvaa sc-gsDJrp bhLkGr gycrnD sc-iCfLBT ccnKaf"
    href="https://twitter.com/fknussel"
    rel="noopener noreferrer"
    target="_blank"
  >
    Twitter
  </a>
  <a
    className="sc-bdvvaa sc-dkPtyc bhLkGr eqBuJi sc-iCfLBT ccnKaf"
    href="https://github.com/fed"
    rel="noopener noreferrer"
    target="_blank"
  >
    GitHub
  </a>
  <a
    className="sc-bdvvaa sc-hKwCoD bhLkGr eQkTzz sc-iCfLBT ccnKaf"
    href="https://medium.com/@fknussel"
    rel="noopener noreferrer"
    target="_blank"
  >
    Medium
  </a>
  <a
    className="sc-bdvvaa sc-eCImvq bhLkGr dXtlEa sc-iCfLBT ccnKaf"
    href="https://codepen.io/fede"
    rel="noopener noreferrer"
    target="_blank"
  >
    CodePen
  </a>
  <a
    className="sc-bdvvaa sc-jRQAMF bhLkGr erPyZN sc-iCfLBT ccnKaf"
    href="https://linkedin.com/in/fknussel"
    rel="noopener noreferrer"
    target="_blank"
  >
    LinkedIn
  </a>
</nav>
`;
